### PR TITLE
Add support for signals on SO3.

### PR DESCRIPTION
**Pull request by ameya98**

### CODE REVIEW

File: e3nn_jax/_src/so3grid.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        assert Rs.shape == (res_theta, res_beta, res_alpha, 3, 3)
+
+        # Evaluate the function at each rotation matrix.
+        func_fn = jax.vmap(jax.vmap(jax.vmap(func)))
+        fs = func_fn(Rs)
+
+        # Move the function output axes to the front.
+        if fs.ndim > 3:
+            for _ in range(fs.ndim - 3):
+                fs = jnp.moveaxis(fs, -1, 0)
```
Comment (mariogeiger) [inline e3nn_jax/_src/so3grid.py:105]: You could consider using `vmap(func, 0, -1)`

File: e3nn_jax/_src/so3grid.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        assert s2_signals.shape == (*batch_dims, res_theta, res_beta, res_alpha)
+        return SO3Signal(s2_signals)
+
+    def integrate_over_angles(self) -> SphericalSignal:
+        delta_theta = self.grid_theta[1] - self.grid_theta[0]
+        return self.s2_signals.replace_values(
+            grid_values=jnp.sum(self.s2_signals.grid_values, axis=-3) * delta_theta
+        )
+
+    def integrate(self) -> SphericalSignal:
```
[the remark this answers is not in the document]
Comment (ameya98) [reply e3nn_jax/_src/so3grid.py:122]: F, will fix! thanks!

File: e3nn_jax/_src/so3grid.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                fs = jnp.moveaxis(fs, -1, 0)
+
+        batch_dims = fs.shape[:-3]
+        assert fs.shape == (*batch_dims, res_theta, res_beta, res_alpha)
+
+        # Account for angle-dependency in Haar measure.
+        fs = fs * (1 - jnp.cos(angles))[..., None, None]
+        s2_signals = s2_signals.replace_values(fs)
+        assert s2_signals.shape == (*batch_dims, res_theta, res_beta, res_alpha)
+        return SO3Signal(s2_signals)
```
Comment (mariogeiger) [inline e3nn_jax/_src/so3grid.py:114]: Seems weird to me that the `SO3Signal` does not contain the actual values of the function but weighted values. Is there a way to change that?

Comment (ameya98) [reply e3nn_jax/_src/so3grid.py:114]: yes, this indeed is a bug which I fixed in https://github.com/e3nn/e3nn-jax/pull/87

**Pull request merged**